### ISSUE TITLE
synonym search

### ISSUE THREAD

**Issue by albertpratsu** [edward-burn/CodelistGeneratorShiny#2]
**Is your feature request related to a problem? Please describe.**
Right now the synonym search seem to only works with synonyms on conditions, but some synonyms also used are hidden in other categories like morphology. For example if I try to search for follicular lymphoma with other names:
1. Search Strategy one of: "Brill - Symmers' disease" or "Follicular lymphosarcoma"
2. Click on Search synonyms
3. Output does not include Follicular non-Hodgkin's lymphoma

**Describe the solution you'd like**
I would like to search synonyms across all vocabularies, not only conditions. But those from other domains should also return related conditions.

**Describe alternatives you've considered**
An alternative would be to add a common synonym database, such as the MeSh Entry Terms . This is probably more extensive than the SNOMED synonyms.

**Comment by edward-burn** [comment]
@albertpratsu I have moved this over to this repository as I think it is specific to the app

For the R package a search can be run against any of the domains in the concept table, for example
```
getCandidateCodes(
  keywords = c("Follicular lymphoma",
               "Brill Symmers' disease", 
               "Follicular lymphosarcoma"),
  domains = c("Condition", 
              "Observation"),
  searchSynonyms = TRUE,
  db = db,
  vocabularyDatabaseSchema = vocabularyDatabaseSchema)
```

However, at the moment the vocabulary underlying the app only has concepts from the condition domain which is why it only returns concepts from that domain.

Adding the full vocabularies to the app is on the to-do list, but in the meantime I recomment using the R package itself

**Comment by albertpratsu** [comment]
@edward-burn I don't have access to a cdm and I'm trying to put the vocabs on a database, so I'll test it soon.

 But, I think that to turn the tool into something more powerful I should be able to search conditions for "Brill Symmers' disease" and get "Follicular non-Hodgkin's lymphoma" without having to know if it's supposed to be a condition or observation as it's synonyms classification by snomed is arbitrary.
 
If we aspire for the tool to be similar to a literature review, maybe the best option is to get in the MeSH entry terms as synonyms irrespectfully of the table we are looking at?

**Comment by edward-burn** [comment]
Yes @albertpratsu you can search in all tables at the same time (specifying the "domains" as all of the ones of interest - it is just that you have the option to only look in one table if you want to narrow the search). 

So in terms of looking at MeSH, these are non-standard codes in the vocabularies. So you can search these by setting [searchNonStandard = TRUE](https://github.com/darwin-eu/CodelistGenerator/blob/5742ab37a5dddffec03eb66cc0ef7a97711ceb18/R/getCandidateCodes.R#L80)

Let's discuss offline, and see with the R package if it already has the functionality required
